.github/workflows/publish-dev-release-to-pypi.yml

```diff
@@ -20,7 +20,7 @@ env:
 
 jobs:
   publish-dev-release-to-pypi:
-    if: github.event.pull_request.merged == true && (startsWith(github.event.pull_request.title, 'fix') || startsWith(github.event.pull_request.title, 'feat'))
+    if: startsWith(github.event.head_commit.message, 'fix') || startsWith(github.event.head_commit.message, 'feat')
```

Comment:
the head commit is the commit created by GitHub when clicking on "Squash & Merge", so the head commit message is the PR title. (Verified with testpypi.org in workflow https://github.com/dasch-swiss/dsp-tools/actions/runs/8244977489)